swagger.yaml

```diff
@@ -1591,7 +1586,22 @@ definitions:
         description: Indicate if VIP DHCP allocation mode is enabled.
         x-nullable: true
         default: false
-
+      http_proxy:
+        type: string
+        description: |
+          A proxy URL to use for creating HTTP connections outside the cluster.
+          http://\<username\>:\<pswd\>@\<ip\>:\<port\>
+        x-nullable: true
+      https_proxy:
```

Comment:
Why do we need extra option for TLS? Can we determine it from the protocol within the URL whie defaulting to http?

Reply:
http-proxy and https-proxy will be assigned to two different configuration entries on OCP Cluster Proxy and before that to http_proxy and https_proxy in the generated ignition files. 

The description of the the fields is taken from https://coreos.com/ignition/docs/latest/configuration-v2_5-experimental.html 

```
proxy (object): options relating to setting an HTTP(S) proxy when fetching resources.
httpProxy (string): will be used as the proxy URL for HTTP requests and HTTPS requests unless overridden by httpsProxy or noProxy.
httpsProxy (string): will be used as the proxy URL for HTTPS requests unless overridden by noProxy.
noProxy (list of strings): specifies a list of strings to hosts that should be excluded from proxying. Each value is represented by an IP address prefix (1.2.3.4), an IP address prefix in CIDR notation (1.2.3.4/8), a domain name, or a special DNS label (*). An IP address prefix and domain name can also include a literal port number (1.2.3.4:80). A domain name matches that name and all subdomains. A domain name with a leading . matches subdomains only. For example foo.com matches foo.com and bar.foo.com; .y.com matches x.y.com but not y.com. A single asterisk (*) indicates that no proxying should be done.
```

Therefore the admin might set a different proxy for http traffic than the https one and we should preserve the option of doing so.